---
title: "RNA_seq_CER7"
author: "BT"
date: "04/12/2018"
output:
  html_document: default
  pdf_document: default
  word_document: default
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

##loading files and packages
```{r}
library("airway")
library("Rsamtools")
library("GenomicFeatures")
library("GenomicAlignments")
library("BiocParallel")
library("Rsubread")
library('DESeq2')
library('pheatmap')
mydir<-file.path('/Users/tangb/general/RNA_seq_CER7/') # containing bam files generated by aligning to genome reference
list.files(mydir)
mycsvfile<-file.path(mydir, 'table.csv')
mysampleTable<-read.csv(mycsvfile,row.names = 1)
mysampleTable
myfilenames<-file.path(mydir, paste0(mysampleTable$run,'_bowtie_aligned.sorted.bam'))
file.exists(myfilenames)
mybamfiles<-BamFileList(myfilenames,yieldSize = 2000000)
seqinfo(mybamfiles[1])
mygtffile<-file.path(mydir,'Magnaporthe_oryzae.MG8.38.gtf')
myfc <- featureCounts(files=myfilenames, 
                    annot.ext=mygtffile, 
                    isGTFAnnotationFile=TRUE,
                    isPairedEnd=TRUE)

```
## alternative way to get matrix
txdb <- makeTxDbFromGFF(mygtffile, format="gtf")
ebg <- exonsBy(txdb, by="gene")
ebg
se <- summarizeOverlaps(features=ebg, 
                        reads=mybamfiles,
                        mode="Union",
                        singleEnd=FALSE,
                        ignore.strand=TRUE,
                        fragments=TRUE ) # omitted by saving time
```{r}
library("GenomicAlignments")
library("BiocParallel")
```

# from this part, we are going to use count matrix to get DEGs with DESeq2.
```{r}
colnames(myfc$counts) <- mysampleTable$run
head(myfc$counts)
countdata<-myfc$count
wt_cer7 <-DESeqDataSetFromMatrix(myfc$count, DataFrame(mysampleTable), ~genotype)
wt_cer7$genotype
help(DESeqDataSetFromMatrix)
wt_cer7<-DESeqDataSet(wt_cer7, design= ~genotype)
countdata_cer7<-assay(wt_cer7)
coldata_cer7<-colData(wt_cer7)
wt_cer7<-wt_cer7[rowSums(assay(wt_cer7)) >=5,  ]
```

create comparison for WT vs CER7, WT vs KO, CER7 vs KO
```{r}
vsd_cer7<-vst(wt_cer7)
class(wt_cer7)
head(colData(vsd_cer7))
colData(vsd_cer7)
plotPCA(vsd_cer7, 'genotype')
alldds<-DESeq(wt_cer7)
res_wt_cer7<-results(alldds, contrast=c('genotype','wt','cer'))
res_wt_rgs<-results(alldds,contrast = c('genotype','wt','rgs'))
res_cer7_rgs<-results(alldds,contrast = c('genotype','cer','rgs'))
mcols(res_cer7_rgs)
mcols(res_wt_rgs)
mcols(res_wt_cer7)
summary(res_cer7_rgs)
summary(res_wt_rgs)
summary(res_wt_cer7)

res_cer7_rgs_DF<-as.data.frame(res_cer7_rgs)
write.csv(res_cer7_rgs_DF,file='cer7_rgs.csv')
res_wt_rgs_DF<-as.data.frame(res_wt_rgs)
write.csv(res_wt_rgs_DF,file='wt_rgs.csv')
res_wt_cer7_DF<-as.data.frame(res_wt_cer7)
write.csv(res_wt_cer7_DF,file='res_wt_cer7.csv')

```
#plotting some plots

```{r pressure, echo=FALSE}
DESeq2::plotMA(res_cer7_rgs, ylim=c(-15,15))
DESeq2::plotMA(res_wt_rgs, ylim=c(-15,15))
DESeq2::plotMA(res_wt_cer7, ylim=c(-15,15))
p<-pheatmap(assay(vsd_cer7), show_rownames = FALSE, show_colnames = FALSE, cellwidth = 15)
```

